tools/analyzer_plugin/tool/doc_generation_utils/generate_rule_docs.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:io';
+
+import 'package:analyzer_plugin/protocol/protocol_common.dart';
+import 'package:markdown/markdown.dart';
+import 'package:over_react_analyzer_plugin/src/doc_utils/contributor_meta_registry.dart';
+import 'package:over_react_analyzer_plugin/src/doc_utils/documented_contributor_meta.dart';
+import 'package:over_react_analyzer_plugin/src/doc_utils/maturity.dart';
+
+import 'generate_contributor_doc_page.dart';
+import 'generate_contributor_type_index.dart';
+import 'shared_html.dart';
+
+class RuleDocGenerator extends ContributorDocPage<DocumentedDiagnosticContributorMeta> {
+  RuleDocGenerator(DocumentedDiagnosticContributorMeta contributor) : super(contributor);
+
+  @override
+  generateAdditionalContributorHeaderSections() => '''
+<p>Severity: ${contributor.severity}</p>
+  ''';
+}
+
+class RulesIndexer extends ContributorIndexer {
+  RulesIndexer() : super(pageTitle: 'OverReact Analyzer Plugin Lints', bodySubtitle: 'Lint Rules');
+
+  static final rules = List.of(DiagnosticMetaRegistry.diagnosticRegistry.meta, growable: false)..sort();
+
+  String get enumerateErrorRules =>
+      rules.where((r) => r.severity == AnalysisErrorSeverity.ERROR).map((r) => '${toDescription(r)}').join('\n\n');
+
+  String get enumerateWarningRules =>
+      rules.where((r) => r.severity == AnalysisErrorSeverity.WARNING).map((r) => '${toDescription(r)}').join('\n\n');
+
+  String get enumerateInfoRules =>
+      rules.where((r) => r.severity == AnalysisErrorSeverity.INFO).map((r) => '${toDescription(r)}').join('\n\n');
+
+  String get enumerateSeverities =>
+      AnalysisErrorSeverity.VALUES.map((severity) => '<li><strong>${severity.name}</strong></li>').join('\n');
+
+  @override
+  String generateContributorsIndexBodySection() => '''
+<h1 id="supported-lints">Supported Lint Rules</h1>
+<p>
+   This list is auto-generated from our sources.
+</p>
+<p>
+   ${markdownToHtml(ruleLeadMatter)}
+</p>
+<p>
+  <ul> 
+    $enumerateSeverities
+  </ul>
+</p>
+<p>
+   ${markdownToHtml(ruleFootMatter)}
+</p>
+
+<h2 id="error-rules">Errors</h2>
+
+   $enumerateErrorRules
+
+<h2 id="warning-rules">Warnings</h2>
+
+   $enumerateWarningRules
+                  
+<h2 id="hint-rules">Hints</h2>
+
+   $enumerateInfoRules
+''';
+}
+
+class OptionsSample {
+  Iterable<DocumentedDiagnosticContributorMeta> rules;
+  OptionsSample(this.rules);
+
+  void generate(String filePath) {
+    var generated = _generate();
+    if (filePath != null) {
+      var outPath = '$filePath/options/index.html';
+      print('Writing to $outPath');
+      File(outPath).writeAsStringSync(generated);
```

Comment:
This fails if the parent dir doesn't exist
```suggestion
      File(outPath)
        ..parent.createSync(recursive: true)
        ..writeAsStringSync(generated);
```